### PR TITLE
compose.each prototype

### PR DESCRIPTION
**Pull request by newtectonics**

### CODE REVIEW

**Comment by newtectonics** [comment]
@iwyrkore @leafyoung Could you take a look at this. It's not ready for production, but this is how I'm imagining some of the compose stuff working, we'll need a pipe, a pipe to specific input properties, and I think we can use this to make historical really powerful.

File: models/credmark/compose/each.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+@describe(
+    slug="compose.each",
+    version="1.0",
+    input=ModelComposeEachDTO,
+    output=ModelComposeEachOutputDTO)
+class ComposeEach(Model):
+    def run(self, input: ModelComposeEachDTO) -> ModelComposeEachOutputDTO:
+        list_model = self.context.run_model(
+            slug=input.model_input_slug,
```
Comment (iwyrkore) [inline models/credmark/compose/each.py:31]: Maybe this should be "fetcher_slug"? Was that the name we decided on for models that do Data Fetch?

File: models/credmark/compose/each.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            input=input.model_input_input,
+            return_type=dict)
+        for key in list_model.keys():
```
Comment (iwyrkore) [inline models/credmark/compose/each.py:34]: Instead of doing this maybe an input param should be the list property name.

**Comment by iwyrkore** [comment]
We might want a way to pass in some extra input to the processor model and combine that with the row data somehow. We'll need to start using this to figure out what we need.

File: models/credmark/compose/each.py (hunk 1 above)
Comment (iwyrkore) [reply models/credmark/compose/each.py:31]: Better to nest: `input.fetch.slug` `input.fetch.input`

```
input = {
   "fetch": {
      "slug": "foo",
      "input": {
          ....
      }
   }
   "process": {
      "slug": "bar",
   }
}
```

**Pull request closed without merging**